worlds/ff6wc/__init__.py

```diff
@@ -340,8 +344,24 @@ def generate_early(self):
                 except ValueError as e:
                     raise ValueError(stesp_err_msg) from e
                 if stesp_min > 0 or stesp_max > 0:
-                    # TODO:
-                    raise NotImplementedError("starting espers is not supported")
+                    # pick a random number of starting espers between min & max specified
+                    num_start_espers = self.random.randint(stesp_min, stesp_max)
+                    chosen_esper_indexes = self.random.sample(range(len(Rom.espers)), num_start_espers)
+                    # update the -sen flag to include a list of chosen esper numbers
+                    sen_flag = ",".join([str(id_) for id_ in chosen_esper_indexes])
+                    # populate list of starting espers
+                    self.starting_espers = [Rom.espers[i] for i in chosen_esper_indexes]
+                    # Now, replace -stesp min max flags with -sen x,y,z,etc
+                    # create a list of flags
+                    splitflags = [flag for flag in self.options.Flagstring.value.split("-")]
+                    # for each flag in the total flagstring
+                    for flag in splitflags:
+                        # find the flag for stesp
+                        if flag.split(" ")[0] == "stesp":
+                            # replace with the specific esper string we built above
+                            splitflags[splitflags.index(flag)] = "sen " + sen_flag + " "
+                        # join the flagstring back together again
+                        self.options.Flagstring.value = "-".join(splitflags) 
```

Comment:
I think I found that this breaks if `stesp` is the LAST flag in the flagstring (in the yaml).
So if the flagstring ends with  ... "-stesp 3 4", and nothing comes after that, it will crash generation.

I would like to make a generic method on the `Flagstring` type to replace one flag with another, to handle this, but I don't know when I would be able to get to that.

Also, the last line here can be dedented, can't it? We don't need to rebuild the flagstring after every iteration of the for loop.

Reply:
I implemented that `replace_flag` function and used it to fix this.